src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -157,6 +157,12 @@ class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: I
     }
   }
 
+  override def reverse = {
+    val r = new ArrayDeque[A](initialSize = size)
+    indices.foreach({i => r.appendAssumingCapacity(this(size - i - 1))})
```

Comment:
I did this: https://github.com/scala/collection-strawman/pull/49/commits/804bf43c60e052c5e2ae4821a821068a72553dc2

Btw, is there a fast way to reverse an array besides looping (something akin to native memcopy)?